junit-platform-commons/src/main/java/org/junit/platform/commons/util/LruCache.java

```diff
@@ -30,8 +32,18 @@
 
 	private final int maxSize;
 
+	/**
+	 * Create a new LRU cache that maintains at most the supplied number of
+	 * entries.
+	 *
+	 * <p>For optimal use of the internal data structures, you should pick a
+	 * number that's one below a power of two since this is based on a
+	 * {@link java.util.HashMap} and the eldest entry will be evicted after
+	 * adding the entry that increases the {@linkplain #size() size} to be above
+	 * {@code maxSize}.
+	 */
```

Comment:
@marcphilipp I can tell that there is a link between powers of two and the fact that this cache extends `HashMap`, but it's not obvious to me what that link is, so I wondered if you could expand the javadoc to clarify things?

Reply:
Well, I could add a `@see HashMap`. Would that help?

Reply:
@marcphilipp Thanks for the suggestion! However, when I search for "two", "2" or "power" in the [javadoc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/HashMap.html) for `HashMap`, I don't get any useful information that clarifies the link between powers of two and hash maps.

I'm sure I'm missing something, so would you mind adding an `@see` pointing to the area in the javadoc that describes the relationship?

Reply:
You're right, it's not really documented, probably on purpose... 😉 

Basically, it uses an internal array of buckets with a length that's always a power of two. For every object, it will compute the hash an put it into one of the buckets. When it needs to resize, it doubles the number of buckets and rehashes all entries.

Reply:
Ahh, okay, got it! :+1: 

It's unclear to me then if adding `@see HashMap` would help. What do you think about adding your description of `HashMap`'s internal resizing policy to the javadoc comment? :)

Reply:
I don't want to describe `HashMap`'s internals in our Javadoc in too much detail. It's an internal API so the comment is mainly there as a reminder. 😉